jsapp/js/universalTable/paginatedQueryUniversalTable.component.tsx

```diff
@@ -8,13 +8,24 @@ import UniversalTable from './universalTable.component';
 import type {UseQueryResult} from '@tanstack/react-query';
 import type {PaginatedResponse} from 'js/dataInterface';
 import type {UniversalTableColumn} from './universalTable.component';
+import type {Record} from 'immutable';
 
-interface PaginatedQueryHook<DataItem> extends Function {
-  (limit: number, offset: number): UseQueryResult<PaginatedResponse<DataItem>>;
+type PaginatedQueryHookData = Record<string, string | number | boolean>;
+
+export type PaginatedQueryHookParams = {
+  limit: number;
+  offset: number;
+} & PaginatedQueryHookData;
+
+interface PaginatedQueryHook<DataItem> {
+  (
+    params: PaginatedQueryHookParams
+  ): UseQueryResult<PaginatedResponse<DataItem>>;
 }
 
 interface PaginatedQueryUniversalTableProps<DataItem> {
   queryHook: PaginatedQueryHook<DataItem>;
+  queryHookData?: PaginatedQueryHookData;
```

Comment:
I would love to see a comment explaining what `queryHookData` is and what is it's main purpose 🫶